src/utils/hdf-serialize.c

```diff
@@ -0,0 +1,1033 @@
+	prop->dpp_entries[9].dpe_type = DAOS_PROP_CO_COMPRESS;
+	prop->dpp_entries[10].dpe_type = DAOS_PROP_CO_ENCRYPT;
+	prop->dpp_entries[11].dpe_type = DAOS_PROP_CO_OWNER;
+	prop->dpp_entries[12].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+	prop->dpp_entries[13].dpe_type = DAOS_PROP_CO_DEDUP;
+	prop->dpp_entries[14].dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+	prop->dpp_entries[15].dpe_type = DAOS_PROP_CO_ALLOCED_OID;
+	prop->dpp_entries[16].dpe_type = DAOS_PROP_CO_ACL;
+
+	entry = &prop->dpp_entries[0];
+	rc = cont_deserialize_prop_str(file_id, entry, "DAOS_PROP_CO_LABEL");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[1];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_LAYOUT_TYPE");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[2];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_LAYOUT_VER");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[3];
+	rc = cont_deserialize_prop_uint(file_id, entry, "DAOS_PROP_CO_CSUM");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[4];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_CSUM_CHUNK_SIZE");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[5];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_CSUM_SERVER_VERIFY");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[6];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_REDUN_FAC");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[7];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_REDUN_LVL");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[8];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_SNAPSHOT_MAX");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[9];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_COMPRESS");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[10];
+	rc = cont_deserialize_prop_uint(file_id, entry, "DAOS_PROP_CO_ENCRYPT");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[11];
+	rc = cont_deserialize_prop_str(file_id, entry, "DAOS_PROP_CO_OWNER");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[12];
+	rc = cont_deserialize_prop_str(file_id, entry,
+				       "DAOS_PROP_CO_OWNER_GROUP");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[13];
+	rc = cont_deserialize_prop_uint(file_id, entry, "DAOS_PROP_CO_DEDUP");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[14];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_DEDUP_THRESHOLD");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[15];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_ALLOCED_OID");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[16];
+	/* read acl as a list of strings in deserialize, then convert back to
+	 * acl for property entry
+ 	 */
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs